r/configure

```diff
@@ -196,20 +196,10 @@ if [ "$ARROW_R_CXXFLAGS" ]; then
 fi
 
 # Test configuration
-CXXCPP="`${R_HOME}/bin/R CMD config CXX11` -E"
-if [ $? -eq 0 ]; then
-  # Great, CXX11 exists for this version of R (current);
-  # now let's set the other two variables
-  CXX11FLAGS=`"${R_HOME}"/bin/R CMD config CXX11FLAGS`
-  CXX11STD=`"${R_HOME}"/bin/R CMD config CXX11STD`
-else
-  # For compatibility with R < 3.4, when these were called CXX1X
-  CXXCPP="`${R_HOME}/bin/R CMD config CXX1X` -E"
-  CXX11FLAGS=`"${R_HOME}"/bin/R CMD config CXX1XFLAGS`
-  CXX11STD=`"${R_HOME}"/bin/R CMD config CXX1XSTD`
-fi
+CXX17FLAGS=`"${R_HOME}"/bin/R CMD config CXX17FLAGS`
```

Comment:
I don't think you actually want/need to delete this `if` block, but regardless the deletion was overzealous and we still need to define `CXXCPP` for the test command on L202. This should fix the Linux R builds (or the reason they're currently failing).

```suggestion
CXXCPP="`${R_HOME}/bin/R CMD config CXX17` -E"
CXX17FLAGS=`"${R_HOME}"/bin/R CMD config CXX17FLAGS`
```

Reply:
Thanks for spotting this!

I'm not sure what the r < 3.4 block would be. Would CXX1X be able to compile c++17?

Reply:
Possibly! We have a crossbow job that will tell us whether it works (if you restore the if block)

Reply:
will do